tools/export.py

```diff
@@ -199,36 +161,60 @@ def stub_roi_feature_extractor(model, extractor_name):
+    data = test_pipeline(data)
+    data = scatter(collate([data], samples_per_gpu=1), [device])[0]
+    return data
+
+
 def main(args):
-    model = init_detector(args.config, args.checkpoint, device='cpu')
-    model.eval().cuda()
     torch.set_default_tensor_type(torch.FloatTensor)
+    model = init_detector(args.config, args.checkpoint, device='cpu')
+    model.eval()
+    model.cuda()
+    device = next(model.parameters()).device
+    cfg = model.cfg
+    fake_data = get_fake_input(cfg, device=device)
 
-    if not args.no_stubs:
+    if args.target == 'openvino' and not args.alt_ssd_export:
         stub_anchor_generator(model, 'rpn_head')
         stub_anchor_generator(model, 'bbox_head')
         stub_roi_feature_extractor(model, 'bbox_roi_extractor')
         stub_roi_feature_extractor(model, 'mask_roi_extractor')
+
+    onnx_model_path = osp.join(args.output_dir,
+                               osp.splitext(osp.basename(args.config))[0] + '.onnx')
     
-    image = np.zeros((128, 128, 3), dtype=np.uint8)
     with torch.no_grad():
-        export(model, image, export_name=args.model, opset=10)
-
-    # add_node_names(args.model)
+        export_to_onnx(model, fake_data, export_name=onnx_model_path, opset=10, alt_ssd_export=args.alt_ssd_export)
+        add_node_names(onnx_model_path)
+        print(f'ONNX model has been saved to "{onnx_model_path}"')
 
-    if args.check:
-        check_onnx_model(args.model)
+    if args.target == 'openvino':
+        input_shape = list(fake_data['img'][0].shape)
+        if args.input_shape:
+            input_shape = [1, 3, *args.input_shape]
+        export_to_openvino(cfg, onnx_model_path, args.output_dir, input_shape)
+    else:
+        check_onnx_model(onnx_model_path)
 
 
 def parse_args():
-    parser = argparse.ArgumentParser(description='Export model to ONNX')
+    parser = argparse.ArgumentParser(description='Export model')
     parser.add_argument('config', help='test config file path')
     parser.add_argument('checkpoint', help="path to file with model's weights")
-    parser.add_argument('model', help='path to output onnx model file')
-    parser.add_argument('--check', action='store_true',
-                        help='check that resulting onnx model is valid')
-    parser.add_argument('--no_stubs', action='store_true',
-                        help='disable all exporting stubs')
+    parser.add_argument('output_dir', help='path to directory to save exported models in')
+
+    subparsers = parser.add_subparsers(title='target', dest='target', help='target model format')
+    parser_onnx = subparsers.add_parser('onnx', help='export to ONNX')
+    parser_openvino = subparsers.add_parser('openvino', help='export to OpenVINO')
+    parser_openvino.add_argument('--input_shape', nargs=2, type=int, default=None,
+                                 help='input shape as a height-width pair')
+    parser_openvino.add_argument('--alt_ssd_export', action='store_true',
```

Comment:
not defined in case of 'onnx' target

Reply:
It is by design. No one needs this argument for ONNX export.

Reply:
https://github.com/opencv/mmdetection/pull/15/files#diff-8c7848a5949a259c04b4eb5347a0d38fR192

Reply:
bad visualization of a link here. 
So
<pre>
export_to_onnx(model, fake_data, export_name=onnx_model_path, opset=10, alt_ssd_export=args.alt_ssd_export)
</pre>

Reply:
I believe, you didn't get the idea.

Two different options are possible:
* Target ONNX. You need your model in ONNX to run it using ONNXRuntime. This way you need a valid ONNX graph (represented by operations from a standard opset).
* Target OpenVINO. You need your model in OpenVINO IR to run it using Inference Engine. This way you have to use ONNX representation as an intermediary to further pass it to Model Optimizer. In this case ONNX graph is not required to be represented in a standard opset, but only be compatible with MO expects to get as input.

So no one needs options for custom opset's configurations if the aim is to obtain only standard opset.

Reply:
I believe, that you did not understand that it does not matter how wonderful your idea is because:
python ./tools/export.py config.py snpashot.pth EXPORT_DIR onnx

  File "./tools/export.py", line 192, in main
    export_to_onnx(model, fake_data, export_name=onnx_model_path, opset=10, alt_ssd_export=args.alt_ssd_export)
AttributeError: 'Namespace' object has no attribute 'alt_ssd_export'

Reply:
Fixed that.